win32/inc/c_pal/job_object_helper.h

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#ifndef JOB_OBJECT_HELPER_H
+#define JOB_OBJECT_HELPER_H
+
+#ifdef __cplusplus
+#include <cstdint>
+#else
+#include <stdint.h>
+#endif
+
+
+#include "macro_utils/macro_utils.h"
+
+#include "umock_c/umock_c_prod.h"
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+    MOCKABLE_FUNCTION(, int, job_object_helper_limit_resources, uint32_t, percent_physical_memory, uint32_t, percent_cpu);
```

Comment:
>job_object_helper_limit_resources [](http://example.com/codeflow?start=29&length=33)

I am going to make a comment (or more) here, which nobody will like, buuuut... 

1) I am hoping that in the big design, these limits are set by the Almighty Resource Governor - and that is SF in the end. Somehow we living in separate process/processes or clumped together should matter differently.

2) Assume we're somehow at our wits end - and everything is "used". Not 1 byte of memory left. The technician comes and wants to ... change the percent physical memory. It is conceivable that this function will fail because it will fail creating the JobObjects. So maybe consider having a handle that keep in memory the JobObject to be used/reused across calls. That is, once the handle is created, adjustment is always possible.

Reply:
Good points. Right now, this is a one-time thing that gets called at service start based on SF config parameters. Something that allows for outside control is a much bigger feature and out of scope.